### PR TITLE
Guided tours removal of tour permissions

### PR DESCRIPTION
**Pull request by obuisard**
Pull Request as a replacement for #40208 and #40220. It compiles some of the corrections made and simplifies the code.

### Summary of Changes

Individual tours have permissions in the original Guided Tours PR.
However, tours don't really require such granularity in most cases and the issues linked to having tours in the asset table outweighs the need for such permissions.
This PR removes permissions for individual tours.

### Testing Instructions

Ideally, create several user profiles with limited permissions and make sure those having limited permissions can't access the functionality they are denied.

### Actual result BEFORE applying this Pull Request

Tours can have individual permissions

### Expected result AFTER applying this Pull Request

Tours and steps only get the general permissions of the component.

### Link to documentations
Please select:
- [x] Documentation link for docs.joomla.org: need to remove the permissions section for tours at [Guided Tours: New or Edit Tour](https://docs.joomla.org/Help4.x:Guided_Tours:_New_or_Edit_Tour)
- [ ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by joomdonation** [comment]
I have tested this item :white_check_mark: successfully on 99f68b1c3a8e5042d842670d83b75e27dbf4fb64<br /><br />Tested new installation, tested update, it is still working well. Try to edit tours, steps, all still work. Did a quick code review, too<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40228">issues.joomla.org/tracker/joomla-cms/40228</a>.</sub>

File: administrator/components/com_admin/sql/updates/mysql/4.3.0-2023-02-15.sql [hunk 1]
```diff
@@ -30,18 +29,18 @@ CREATE TABLE IF NOT EXISTS `#__guidedtours` (
...
- (10, 0, 'COM_GUIDEDTOURS_TOUR_SMARTSEARCH_TITLE', 'COM_GUIDEDTOURS_TOUR_SMARTSEARCH_DESCRIPTION', 10, '[\"*\"]', 'administrator/index.php?option=com_finder&view=filters', CURRENT_TIMESTAMP(), 0, CURRENT_TIMESTAMP(), 0, NULL, NULL, 1, '*', 1),
- (11, 0, 'COM_GUIDEDTOURS_TOUR_USERS_TITLE', 'COM_GUIDEDTOURS_TOUR_USERS_DESCRIPTION', 11, '[\"*\"]', 'administrator/index.php?option=com_users&view=users', CURRENT_TIMESTAMP(), 0, CURRENT_TIMESTAMP(), 0, NULL, NULL, 1, '*', 1);
+INSERT IGNORE INTO `#__guidedtours` (`id`, `title`, `description`, `ordering`, `extensions`, `url`, `created`, `created_by`, `modified`, `modified_by`, `checked_out_time`, `checked_out`, `published`, `language`, `access`) VALUES
+ (1, 'COM_GUIDEDTOURS_TOUR_GUIDEDTOURS_TITLE', 'COM_GUIDEDTOURS_TOUR_GUIDEDTOURS_DESCRIPTION', 1, '[\"com_guidedtours\"]', 'administrator/index.php?option=com_guidedtours&view=tours', CURRENT_TIMESTAMP(), 0, CURRENT_TIMESTAMP(), 0, NULL, NULL, 1, '*', 1),
```
Comment (Quy) [inline administrator/components/com_admin/sql/updates/mysql/4.3.0-2023-02-15.sql:33]: Is it necessary to escape the double quote `'[\"com_guidedtours\"]'`?

Comment (richard67) [reply administrator/components/com_admin/sql/updates/mysql/4.3.0-2023-02-15.sql:33]: For the database (MySQL / MariaDB or PostgreSQL) it doesn't need that. No idea if the guided tours component wants that for some reason.

Comment (richard67) [reply administrator/components/com_admin/sql/updates/mysql/4.3.0-2023-02-15.sql:33]: But it is like that already without this PR here for the extensions column in the tours table so I would say it's something to be fixed independently from this PR.

**Comment by khu5h1** [comment]
I have tested this item :white_check_mark: successfully on 99f68b1c3a8e5042d842670d83b75e27dbf4fb64<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40228">issues.joomla.org/tracker/joomla-cms/40228</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40228">issues.joomla.org/tracker/joomla-cms/40228</a>.</sub>

**Pull request merged**